host-go/runtimes/wasmtime/runtime.go

```diff
@@ -0,0 +1,139 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package wasmtime
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/lens-vm/lens/host-go/engine/module"
+	"github.com/lens-vm/lens/host-go/engine/pipes"
+
+	"github.com/bytecodealliance/wasmtime-go/v14"
+)
+
+type wRuntime struct {
+	store *wasmtime.Store
+}
+
+var _ module.Runtime = (*wRuntime)(nil)
+
+func New() module.Runtime {
+	engine := wasmtime.NewEngine()
+	store := wasmtime.NewStore(engine)
+
+	return &wRuntime{
+		store: store,
+	}
+}
+
+type wModule struct {
+	rt     *wRuntime
+	module *wasmtime.Module
+}
+
+var _ module.Module = (*wModule)(nil)
+
+func (rt *wRuntime) NewModule(wasmBytes []byte) (module.Module, error) {
+	module, err := wasmtime.NewModule(rt.store.Engine, wasmBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &wModule{
+		rt:     rt,
+		module: module,
+	}, nil
+}
+
+func (m *wModule) NewInstance(functionName string, paramSets ...map[string]any) (module.Instance, error) {
+	instance, err := wasmtime.NewInstance(m.rt.store, m.module, []wasmtime.AsExtern{})
+	if err != nil {
+		return module.Instance{}, err
+	}
+
+	mem := instance.GetExport(m.rt.store, "memory")
+	if mem == nil {
+		return module.Instance{}, errors.New(fmt.Sprintf("Export `%s` does not exist", "memory"))
```

Comment:
suggestion: If not in a separate file, moving this to a `NewExportError` function would be nicer than to have the same string formatting implementation multiple times.

Reply:
Yes I considered that, but proper standarization is a bit more involved than cut-pasting these into consts, I one of the runtimes returns its own errors instead of these.  I'd want to do them all together in a dedicated ticket, with tests.